Land stubs for the Screen Wake Lock API

cdumez opened the pull request:
#### 4500a5a19d5349da167b2adff6b9d32edb097f01
<pre>
Land stubs for the Screen Wake Lock API
<a href="https://bugs.webkit.org/show_bug.cgi?id=245551">https://bugs.webkit.org/show_bug.cgi?id=245551</a>

Reviewed by Brent Fulgham.

Land stubs for the Screen Wake Lock API:
- <a href="https://www.w3.org/TR/screen-wake-lock">https://www.w3.org/TR/screen-wake-lock</a>

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/screen-wake-lock/Navigator+ScreenWakeLock.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.cpp: Added.
(WebCore::NavigatorScreenWakeLock::NavigatorScreenWakeLock):
(WebCore::NavigatorScreenWakeLock::from):
(WebCore::NavigatorScreenWakeLock::supplementName):
(WebCore::NavigatorScreenWakeLock::wakeLock):
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp: Added.
(WebCore::WakeLock::request):
* Source/WebCore/Modules/screen-wake-lock/WakeLock.h: Added.
(WebCore::WakeLock::create):
* Source/WebCore/Modules/screen-wake-lock/WakeLock.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.cpp: Added.
(WebCore::WakeLockSentinel::WakeLockSentinel):
(WebCore::WakeLockSentinel::release):
(WebCore::WakeLockSentinel::activeDOMObjectName const):
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockType.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockType.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: <a href="https://commits.webkit.org/254825@main">https://commits.webkit.org/254825@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d6997669744a3e1a5d66ba6c23be5822e7020401

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90355 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34942 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20959 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99690 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/157157 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33440 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82715 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/96137 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96010 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26570 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77206 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26446 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81381 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81156 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69452 "Passed tests") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/81939 "Found 1 new JSC stress test failure: wasm.yaml/wasm/lowExecutableMemory/imports-oom.js.default-wasm (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/34536 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15211 "Found 1 new test failure: css3/scroll-snap/scroll-snap-drag-scrollbar-thumb-with-relayouts.html (failure)") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/76876 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32360 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16168 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26584 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36123 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39123 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/79462 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1460 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38037 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35257 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17424 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on LayoutTests/imported/w3c/web-platform-tests/screen-wake-lock/idlharness.https.window-expected.txt:
```diff
@@ -11,33 +11,33 @@ PASS Navigator includes NavigatorContentUtils: member names are unique
+PASS WakeLockSentinel interface: attribute released
+PASS WakeLockSentinel interface: attribute type
+PASS WakeLockSentinel interface: operation release()
+PASS WakeLockSentinel interface: attribute onrelease
```
:-D

brentfulgham commented on LayoutTests/imported/w3c/web-platform-tests/screen-wake-lock/wakelock-insecure-context.any-expected.txt:
```diff
@@ -1,3 +1,3 @@
 
-PASS Wake Lock API is not exposed in an insecure context
+FAIL Wake Lock API is not exposed in an insecure context assert_false: 'WakeLock' must not be exposed expected false got true
```
Do we need a radar to adopt Secure Contexts in more places?

brentfulgham replied:
Weird -- so we declare this is Secure Context in IDL, but it's still exposed in insecure? That's something we should flag as a separate bug...

cdumez replied:
You'll notice that we have a FAIL here but GTK has a PASS. Our implementation is correct. The reason we fail is that we run WPT tests incorrectly on the Cocoa ports. We're still running them over localhost instead of the web-platform.test domain. As a side effect, localhost is treated as a secure context, even when not HTTPS.
The test doesn't expect to be run over localhost, assumes the context shouldn't be secure, but it is for us. This is why we're failing this.

cdumez replied:
And we already have a radar to fix the fact that we should run WPT tests over web-platform.test domain instead of localhost. I even prepared a WIP PR at https://github.com/WebKit/WebKit/pull/4438. However, EWS is failing some loads due to SSL certificate failures and I haven't been able to figure them out yet. I cannot reproduce those failures locally so it doesn't help.

brentfulgham reviewed (approved):
LGTM. r=me

cdumez committed:
Land stubs for the Screen Wake Lock API
https://bugs.webkit.org/show_bug.cgi?id=245551

Reviewed by Brent Fulgham.

Land stubs for the Screen Wake Lock API:
- https://www.w3.org/TR/screen-wake-lock

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/screen-wake-lock/Navigator+ScreenWakeLock.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.cpp: Added.
(WebCore::NavigatorScreenWakeLock::NavigatorScreenWakeLock):
(WebCore::NavigatorScreenWakeLock::from):
(WebCore::NavigatorScreenWakeLock::supplementName):
(WebCore::NavigatorScreenWakeLock::wakeLock):
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp: Added.
(WebCore::WakeLock::request):
* Source/WebCore/Modules/screen-wake-lock/WakeLock.h: Added.
(WebCore::WakeLock::create):
* Source/WebCore/Modules/screen-wake-lock/WakeLock.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.cpp: Added.
(WebCore::WakeLockSentinel::WakeLockSentinel):
(WebCore::WakeLockSentinel::release):
(WebCore::WakeLockSentinel::activeDOMObjectName const):
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.idl: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockType.h: Added.
* Source/WebCore/Modules/screen-wake-lock/WakeLockType.idl: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: https://commits.webkit.org/254825@main
